polkadot/runtime/rococo/src/genesis_config_presets.rs

```diff
@@ -62,39 +47,64 @@ fn get_authority_keys_from_seed(
 	BeefyId,
 ) {
 	let keys = get_authority_keys_from_seed_no_beefy(seed);
-	(keys.0, keys.1, keys.2, keys.3, keys.4, keys.5, keys.6, get_from_seed::<BeefyId>(seed))
+	(
+		keys.0,
+		keys.1,
+		keys.2,
+		keys.3,
+		keys.4,
+		keys.5,
+		keys.6,
+		BeefyId::from(
+			sp_consensus_beefy::ecdsa_crypto::Pair::from_string(&format!("//{}", seed), None)
+				.expect("should parse str seed to keyring")
+				.public(),
+		),
+	)
 }
 
 /// Helper function to generate stash, controller and session key from seed
 fn get_authority_keys_from_seed_no_beefy(
 	seed: &str,
 ) -> (AccountId, AccountId, BabeId, GrandpaId, ValidatorId, AssignmentId, AuthorityDiscoveryId) {
+	use core::str::FromStr;
 	(
-		get_account_id_from_seed::<sr25519::Public>(&format!("{}//stash", seed)),
-		get_account_id_from_seed::<sr25519::Public>(seed),
-		get_from_seed::<BabeId>(seed),
-		get_from_seed::<GrandpaId>(seed),
-		get_from_seed::<ValidatorId>(seed),
-		get_from_seed::<AssignmentId>(seed),
-		get_from_seed::<AuthorityDiscoveryId>(seed),
+		Sr25519Keyring::from_str(&format!("{}//stash", seed))
+			.expect("should parse str seed to keyring")
+			.to_account_id(),
+		Sr25519Keyring::from_str(seed)
+			.expect("should parse str seed to keyring")
+			.to_account_id(),
+		BabeId::from(
+			Sr25519Keyring::from_str(seed)
+				.expect("should parse str seed to keyring")
+				.public(),
+		),
+		GrandpaId::from(
+			Ed25519Keyring::from_str(seed)
+				.expect("should parse str seed to keyring")
+				.public(),
+		),
+		ValidatorId::from(
+			Sr25519Keyring::from_str(seed)
+				.expect("should parse str seed to keyring")
+				.public(),
+		),
+		AssignmentId::from(
+			Sr25519Keyring::from_str(seed)
+				.expect("should parse str seed to keyring")
+				.public(),
+		),
+		AuthorityDiscoveryId::from(
+			Sr25519Keyring::from_str(seed)
+				.expect("should parse str seed to keyring")
+				.public(),
```

Comment:
Not sure if we want to have it in this form. `get_from_seed` was compact and more readable. Now we will have a lot of repetition, my personal feeling is that it is not better from what have now.

If the copy of this function is spread across the codebase - maybe we should keep the function and move it to some low-level crate?

Reply:
That's why I don't make PR ready even though all get_account_id_from_seed / get_from_seed removed. It looks not clean and readable. I am trying to rewrite this part. If can't avoid repetition, maybe we should consider keeping the function.

Reply:
> Not sure if we want to have it in this form. `get_from_seed` was compact and more readable. With `get_seed` removed we will have a lot of repetition of this `AuthorityDiscoveryId::from...` (and variants) combo. My personal feeling is that it is not better from what have now.
> 
> If the copy of this function is spread across the codebase - maybe we should keep the function and move it to some low-level crate?

That's why I don't make PR ready even though all get_account_id_from_seed / get_from_seed removed. It looks not clean and readable. I am trying to rewrite this part. If can't avoid repetition, maybe we should consider keeping the function.